### PR TITLE
LEAN-3406: Modify the save project API to delete any existing data.

### PR DESCRIPTION
**Pull request by mbetamony**

### CODE REVIEW

File: src/DomainServices/ProjectService.ts [hunk 1]
```diff
@@ -430,4 +420,22 @@ export class ProjectService {
...
+      }
+
+      const errorMessage = validate(updatedDoc)
+      if (errorMessage) {
+        throw new SyncError(errorMessage, updatedDoc)
+      }
+
+      return updatedDoc
+    })
+  }
```
Comment (mbetamony) [inline src/DomainServices/ProjectService.ts:440]: The original method here in [ContainerService](https://github.com/Atypon-OpenSource/manuscripts-api/blob/4f4f885f39f265510378d90b81818ba671de4a7f/src/DomainServices/Container/ContainerService.ts#L729-L731)  updated the manuscriptID. Let me know if we need that, I did some debugging and it seems the models being sent already have the manuscriptID.
@mnatsheh, @hhusban

Comment (mnatsheh) [reply src/DomainServices/ProjectService.ts:440]: Is this logic the same as the one in https://github.com/Atypon-OpenSource/manuscripts-api/blob/82997e54982c8b46b01ebf8a65003b56150c5939/src/DomainServices/ProjectService.ts#L127 ?

Comment (mnatsheh) [reply src/DomainServices/ProjectService.ts:440]: > Technically yes, we might want to only update the updatedAt and validate the model in here, right?

That's fine. We should remove the validation from the `map` call and use `validate` with all the models (same as `importJats`).

We don't need to have this function `async` as far as I can tell.

Comment (mbetamony) [reply src/DomainServices/ProjectService.ts]: Done, I don't think QA testing is needed here, i'll merge directly to master if that's OK.

**Review by mnatsheh — APPROVED** [review]
I'm ok with not explicitly testing this, as long as you made sure the FE works.

**Comment by mbetamony** [comment]
> I'm ok with not explicitly testing this, as long as you made sure the FE works.

Seems to be working fine, I had to add createdAt back to the models.

**Pull request merged**